build/external.cmake

```diff
@@ -0,0 +1,55 @@
+include(ExternalProject)
+set(EXTERNAL_BASE_DIR ${CMAKE_BINARY_DIR}/external CACHE STRING "External base directory")
+
+ExternalProject_Add(
+    zlib_src
+    PREFIX ${EXTERNAL_BASE_DIR}
+    GIT_REPOSITORY https://github.com/madler/zlib.git
+    GIT_TAG 51b7f2abdade71cd9bb0e7a373ef2610ec6f9daf # v1.3.1
+    GIT_SHALLOW True
+    UPDATE_DISCONNECTED True
+    CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${EXTERNAL_BASE_DIR} -DCMAKE_BUILD_TYPE=Release -DCMAKE_MSVC_RUNTIME_LIBRARY=${CMAKE_MSVC_RUNTIME_LIBRARY} -DZLIB_BUILD_MINIZIP=1
+    BUILD_COMMAND ${CMAKE_COMMAND} --build . --config Release
+    BUILD_BYPRODUCTS ${EXTERNAL_BASE_DIR}/lib/libz.a
+    INSTALL_COMMAND ${CMAKE_COMMAND} --install . --config Release
+)
+add_library(zlib STATIC IMPORTED)
+set_target_properties(zlib PROPERTIES IMPORTED_LOCATION ${EXTERNAL_BASE_DIR}/lib/libz.a)
+add_dependencies(zlib zlib_src)
+
+set(ZLIB_SRC_DIR ${EXTERNAL_BASE_DIR}/src/zlib_src)
+set_source_files_properties(${ZLIB_SRC_DIR}/contrib/minizip/ioapi.c PROPERTIES GENERATED 1)
+set_source_files_properties(${ZLIB_SRC_DIR}/contrib/minizip/unzip.c PROPERTIES GENERATED 1)
+set_source_files_properties(${ZLIB_SRC_DIR}/contrib/minizip/iowin32.c PROPERTIES GENERATED 1)
+
+ExternalProject_Add(
+    xml2_src
+    PREFIX ${EXTERNAL_BASE_DIR}
+    GIT_REPOSITORY https://github.com/GNOME/libxml2.git
+    GIT_TAG 60d3056c97067e6cb2125284878ed7c99c90ed81 # v2.13.4
```

Comment:
```suggestion
    GIT_TAG v2.13.4
```

Reply:
Per cmake document [ExternalProject](https://cmake.org/cmake/help/latest/module/ExternalProject.html), commit hash should be preferred than tag.
Anyway, if you really want tag, I will update it.

Reply:
OK to keep as is.